src/components/pages/DonationForm.vue

```diff
@@ -1,19 +1,34 @@
 <template>
 	<div id="laika-donation">
 		<keep-alive>
-			<component
-				ref="currentPage"
-				:is="currentFormComponent"
-				v-on:next-page="changePageIndex( 1 )"
-				v-on:previous-page="changePageIndex( -1 )"
-				v-bind="currentProperties">
-			</component>
+			<PaymentPage
+				v-if="currentPageIndex === 0"
+				@next-page="currentPageIndex = 1"
+				:assets-path="assetsPath"
+				:payment-amounts="paymentAmounts"
+				:payment-intervals="paymentIntervals"
+				:payment-types="paymentTypes"
+			/>
+			<AddressPage
+				v-else
+				@previous-page="currentPageIndex = 0"
+				:assets-path="assetsPath"
+				:validate-addressUrl="validateAddressUrl"
+				:validate-email-url="validateEmailUrl"
+				:validate-bank-data-url="validateBankDataUrl"
+				:validate-legacy-bank-data-url="validateLegacyBankDataUrl"
+				:countries="countries"
+				:salutations="salutations"
+				:tracking-data="trackingData"
+				:campaign-values="campaignValues"
+				:address-validation-patterns="addressValidationPatterns"
+			/>
 		</keep-alive>
```

Comment:
I like that it's much more readable in the template now with the different props for different pages etc.

Reply:
Yeah and it allows us to A/B test the forms easier. I was worried the `keep-alive` tag wouldn't work with 2 components and a `v-if` but it seems to be fine.